jormungandr-integration-tests/src/mock/client.rs

```diff
@@ -0,0 +1,233 @@
+extern crate base64;
+extern crate chain_impl_mockchain;
+extern crate futures;
+extern crate futures_cpupool;
+extern crate grpc;
+extern crate hex;
+extern crate protobuf;
+
+use crate::mock::{
+    grpc::*,
+    proto::{node::*, node_grpc::*},
+    read_into,
+};
+use chain_core::property::FromStr;
+use chain_core::property::Serialize;
+use chain_impl_mockchain as chain;
+use chain_impl_mockchain::{
+    block::{Block as ChainBlock, Header as ChainHeader},
+    fragment::Fragment as ChainFragment,
+    key::Hash,
+};
+use protobuf::RepeatedField;
+
+pub struct JormungandrClient {
+    client: NodeClient,
+    host: String,
+    port: u16,
+}
+
+impl Clone for JormungandrClient {
+    fn clone(&self) -> Self {
+        JormungandrClient::new(&self.host, self.port)
+    }
+}
+
+impl JormungandrClient {
+    pub fn new(host: &str, port: u16) -> Self {
+        let client_conf = Default::default();
+        let client = NodeClient::new_plain(host, port, client_conf).unwrap();
+        Self {
+            client: client,
+            host: host.to_owned(),
+            port: port,
+        }
+    }
+
+    pub fn get_blocks(&self, blocks_id: &Vec<Hash>) -> Vec<ChainBlock> {
+        self.get_blocks_stream(blocks_id)
+            .into_future()
+            .wait_drop_metadata()
+            .unwrap()
+            .into_iter()
+            .map(|x| read_into(x.get_content()))
+            .collect()
+    }
+
+    fn get_blocks_stream(&self, blocks_id: &Vec<Hash>) -> grpc::StreamingResponse<Block> {
+        let block_ids_u8 = blocks_id
+            .iter()
+            .map(|x| x.as_ref().iter().cloned().collect())
+            .collect::<Vec<Vec<u8>>>();
+        let mut block_id = BlockIds::new();
+        block_id.set_ids(RepeatedField::from_vec(block_ids_u8));
+        self.client
+            .get_blocks(grpc::RequestOptions::new(), block_id)
+    }
+
+    pub fn get_headers(&self, blocks_id: &Vec<Hash>) -> Vec<ChainHeader> {
+        self.get_headers_stream(blocks_id)
+            .into_future()
+            .wait_drop_metadata()
+            .unwrap()
+            .into_iter()
+            .map(|x| read_into(x.get_content()))
+            .collect()
```

Comment:
You could try to begin writing macros to save this much typing.

Also maybe, a generic wrapper that encapsulates the response:

```rust
pub struct StreamingResponse<T> {
    inner: grpc::StreamingResponse<T>,
}
```

The wrapper would then have accessor methods for getting the streamed data in a preferred way.

```rust
impl<T> StreamResponse<T> {
    pub fn to_vec(self) -> Vec<T> {
        self.into_iter()
            .map(|x| read_into(x.get_content()))
            .collect()
    }
}
```

Reply:
I used macros